modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationUtil.java

```diff
@@ -886,6 +889,124 @@ public static void touch(DynamicConfiguration<?, ?> cfg) {
         }
     }
 
+    /**
+     * Maps iterable via provided mapper function. Filter all {@code null} mapped values.
+     *
+     * @param iterable Basic iterable.
+     * @param mapper Conversion function.
+     * @param <T1> Base type of the iterable.
+     * @param <T2> Type for view.
+     * @return Mapped iterable.
+     */
+    public static <T1, T2> Iterable<T2> mapIterable(
+            @Nullable Iterable<? extends T1> iterable,
+            @Nullable Function<? super T1, ? extends T2> mapper
+    ) {
+        if (iterable == null) {
+            return Collections.emptyList();
+        }
+
+        if (mapper == null) {
+            return (Iterable<T2>) iterable;
+        }
+
+        return () -> {
+            Iterator<? extends T1> innerIterator = iterable.iterator();
+            return new Iterator<>() {
+                @Nullable
+                private T2 next;
+
+                @Override
+                public boolean hasNext() {
```

Comment:
What if you call `hasNext()` several times? It should work, but it won't. Please implement the iterator properly